src/tests/unit_x509.cpp

```diff
@@ -1592,8 +1593,885 @@ Test::Result test_x509_tn_auth_list_extension_decode() {
    return result;
 }
 
+Test::Result test_x509_ip_addr_blocks_extension_decode() {
+   Test::Result result("X509 IP Address Block decode");
+   result.start_timer();
+   const std::string filename("IPAddrBlocksAll.pem");
+
+   Botan::X509_Certificate cert(Test::data_file("x509/x509test/" + filename));
+
+   using Botan::Cert_Extension::IPAddressBlocks;
+
+   auto ip_addr_blocks = cert.v3_extensions().get_extension_object_as<IPAddressBlocks>();
+
+   const auto& addr_blocks = ip_addr_blocks->addr_blocks();
+   result.confirm("cert has IPAddrBlocks extension", ip_addr_blocks != nullptr, true);
+   result.confirm("cert has exactly one IpAddrBlock", addr_blocks.size() == 2, true);
```

Comment:
This should use `test_eq(addr_blocks.size(), 2)`

Also the "has exactly one" is IIUC incorrect